test/testdata/resolver/mixes_in_class_methods.rb

```diff
@@ -21,11 +21,15 @@ class Test
 Test.mixin_class_method
 Test.new.mixin_method
 
+module Timesout
+  extend T::Helpers
+  mixes_in_class_methods(Mixin)
+end
```

Comment:
@elliottt I made the necessary changes and came across this timeout behaviour involving a recursive `mixes_in_class_methods` call. This only occurs if mixed in module (`Mixin`) is also included in another class (`Test`).

I couldn't figure out why including triggers this behaviour. This timeout is introduced in this PR.

Could it be a side effect of the `rebind` usage?

Reply:
Compiled with `sanitize` and got a stackoverflow on https://github.com/sorbet/sorbet/blob/master/core/Names.cc#L176

Looks like there is an infinite loop involving this call within`enterClassSymbol` https://github.com/sorbet/sorbet/blob/master/core/GlobalState.cc#L943 and this recursive call https://github.com/sorbet/sorbet/blob/master/core/Names.cc#L185

Reply:
Here's a minimal example that reproduces the timeout that you're seeing:
```ruby
# typed: true

module M
  extend T::Helpers
  module Cls; end
  mixes_in_class_methods Cls
end

class T1
  include M
end

class T2
  mixes_in_class_methods M
end
```

Reply:
Here's an even smaller one:
```ruby
# typed: true

module M
  extend T::Helpers
  module Cls; end
  mixes_in_class_methods Cls
end

class T1
  include M
  mixes_in_class_methods M
end
```

Reply:
Walking through the changes in GlobalState.cc in the debugger, the first time it hits T1 it creates a singleton for it. When that singleton is found in a subsequent iteration, it has a singleton created and it just keeps going like this until the name it's using to create the singleton classes overflows.

I think there's something funny going on with the interaction of include here, but I'll need to dig deeper to make sense of that.